### PR TITLE
feat(actions): precise that smart action middleware is mandatory

### PR DESCRIPTION
**Pull request by arnaudbesnier**

### CODE REVIEW

File: getting-started/setup-guide.md [hunk 1]
```diff
@@ -140,7 +152,9 @@ class Forest::OrdersController < ForestLiana::SmartActionsController
 end
 ```
 
-Note that Forest Admin takes care of the authentication thanks to the `ForestLiana::SmartActionsController` parent class controller.
+{% hint style="warning" %}
+You must make sure that all your Smart Actions controllers extend from the `ForestLiana::SmartActionsController`. This is mandatory to ensure that all features built on top of Smart Actions work as expected (authentication, approval workflows,...)
```
Comment (Thenkei) [inline getting-started/setup-guide.md:156]:
```suggestion
You must make sure that all your Smart Actions controllers extend from the `ForestLiana::SmartActionsController`. This is mandatory to ensure that all features built on top of Smart Actions work as expected (authentication, permissions, approval workflows,...)
```

File: getting-started/setup-guide.md [hunk 2]
```diff
@@ -66,6 +66,12 @@ router.post('/actions/refund-order', permissionMiddlewareCreator.smartAction(),
 
 module.exports = router;
 ```
+
+{% hint style="warning" %}
+You must make sure that all your Smart Actions routes are configured with the Smart Action middleware:
+`permissionMiddlewareCreator.smartAction()`. This is mandatory to ensure that all features built on top of Smart Actions work as expected.
```
Comment (Thenkei) [inline getting-started/setup-guide.md:72]:
```suggestion
`permissionMiddlewareCreator.smartAction()`. This is mandatory to ensure that all features built on top of Smart Actions work as expected (permissions, approval workflows,...).
```

File: getting-started/setup-guide.md (hunk 1 above)
Comment (arnaudbesnier) [reply getting-started/setup-guide.md]: Fun fact, I added it in the before your comment :)

File: getting-started/setup-guide.md [hunk 3]
```diff
@@ -97,6 +103,12 @@ router.post('/actions/refund-order', permissionMiddlewareCreator.smartAction(),
...
```
Comment (Thenkei) [inline getting-started/setup-guide.md:109]:
```suggestion
`permissionMiddlewareCreator.smartAction()`. This is mandatory to ensure that all features built on top of Smart Actions work as expected (permissions, approval workflows,...).
```

**Review by Thenkei — COMMENTED** [review]
Let's make the explanation less evasive like you've done for rails. :)

File: getting-started/setup-guide.md (hunk 1 above)
Comment (Thenkei) [reply getting-started/setup-guide.md]: I just saw it 🎯

File: reference-guide/actions/create-and-manage-smart-actions/README.md [hunk 4]
```diff
@@ -160,6 +160,12 @@ router.post('/actions/mark-as-live', permissionMiddlewareCreator.smartAction(),
 module.exports = router;
 ```
 {% endcode %}
+
+{% hint style="warning" %}
+You must make sure that all your Smart Actions routes are configured with the Smart Action middleware:
+`permissionMiddlewareCreator.smartAction()`. This is mandatory to ensure that all features built on top of Smart Actions work as expected.
```
Comment (Thenkei) [inline reference-guide/actions/create-and-manage-smart-actions/README.md:166]: Same comment

File: reference-guide/actions/create-and-manage-smart-actions/README.md [hunk 5]
```diff
@@ -180,6 +186,12 @@ router.post('/actions/mark-as-live', permissionMiddlewareCreator.smartAction(),
...
```
Comment (Thenkei) [inline reference-guide/actions/create-and-manage-smart-actions/README.md:192]: Same comment :)

**Review by Thenkei — APPROVED** [review]
Let's merge this version. It's better than before. I'm not sure we need to be so much precise after all.

**Pull request merged**